benchexec/container.py

```
@@ -725,29 +805,165 @@ def make_overlay_mount(mount, lower, upper, work):
+            return [int(part) for part in match[1].split(".")]
+        else:
+            logging.warning("Could not find version information of %s in output.", fuse)
+            return None
+    except subprocess.CalledProcessError:
+        return None
+
+
+def setup_fuse_overlay(temp_base, work_base):
+    """
+    Check if fuse-overlayfs is available on the system and,
+    if so, creates a temporary overlay filesystem by stacking the root directory
+    with a specified temporary directory.
+
+    @return: The path to the mounted overlay filesystem if successful, None otherwise.
+    """
+    fuse = shutil.which("fuse-overlayfs")
+    if fuse is None:
+        return None
+    temp_fuse = temp_base + b"/fuse"
+    work_fuse = work_base + b"/fuse"
+    os.makedirs(temp_fuse, exist_ok=True)
+    os.makedirs(work_fuse, exist_ok=True)
+
+    logging.debug(
+        "Creating overlay mount with %s: target=%s, lower=%s, upper=%s, work=%s",
+        fuse,
+        temp_fuse,
+        b"/",
+        temp_base,
+        work_fuse,
+    )
+
+    cmd = (
+        fuse,
+        b"-o",
+        b"lowerdir=/"
+        + b",upperdir="
+        + escape_overlayfs_parameters(temp_base)
+        + b",workdir="
+        + escape_overlayfs_parameters(work_fuse),
+        escape_overlayfs_parameters(temp_fuse),
+    )
+
+    try:
+        with permitted_cap_as_ambient():
+            # Temporarily elevate permitted capabilities to the inheritable set
+            # and raise them in the ambient set.
+            result = subprocess.run(
+                args=cmd,
+                check=True,
+                stdin=subprocess.DEVNULL,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+            if result.stdout:
+                logging.debug("fuse-overlayfs: %s", result.stdout.decode())
+        return temp_fuse
+    except subprocess.CalledProcessError:
+        return None
```

I think here we should not hide the error completely. Let's log it?